Source/WebCore/css/StyleProperties.cpp

```diff
@@ -71,8 +71,11 @@ String serializeLonghandValue(const CSS::SerializationContext& context, CSSPrope
     case CSSPropertyStrokeOpacity:
         // FIXME: Handle this when creating the CSSValue for opacity, to be consistent with other CSS value serialization quirks.
         // Opacity percentage values serialize as a fraction in the range 0-1, not "%".
-        if (auto* primitive = dynamicDowncast<CSSPrimitiveValue>(value); primitive && primitive->isPercentage())
+        if (auto* primitive = dynamicDowncast<CSSPrimitiveValue>(value); primitive && primitive->isPercentage()) {
+            if (primitive->isCalculated())
+                return makeString("calc("_s, primitive->resolveAsPercentageDeprecated() / 100, ")"_s);
```

Comment:
this assumes that all calculated values are `calc()` functions.

what about other math functions? for ex. `opacity: clamp(..);` will this return `calc()`?

Reply:
...i'm curious how you're passing test for other function types also